### ISSUE TITLE
ios build err  version 1.41.0

### ISSUE THREAD

**Issue by qq805005062** [libuv/libuv#3207]
```
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:190:28: error: a parameter list without types is only
      allowed in a function definition
  void (*pIOObjectRelease)(io_object_t);
                           ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:192:3: warning: implicit declaration of function
      'CFMutableDictionaryRef' is invalid in C99 [-Wimplicit-function-declaration]
  CFMutableDictionaryRef (*pIOServiceMatching)(const char*);
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:192:28: error: use of undeclared identifier
      'pIOServiceMatching'
  CFMutableDictionaryRef (*pIOServiceMatching)(const char*);
                           ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:192:48: error: expected expression
  CFMutableDictionaryRef (*pIOServiceMatching)(const char*);
                                               ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:194:50: warning: type specifier missing, defaults to 'int'
      [-Wimplicit-int]
                                                 CFMutableDictionaryRef,
                                                 ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:195:50: error: unknown type name 'io_iterator_t'
                                                 io_iterator_t*);
                                                 ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:196:3: warning: implicit declaration of function
      'io_service_t' is invalid in C99 [-Wimplicit-function-declaration]
  io_service_t (*pIOIteratorNext)(io_iterator_t);
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:196:18: error: use of undeclared identifier
      'pIOIteratorNext'
  io_service_t (*pIOIteratorNext)(io_iterator_t);
                 ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:196:35: error: use of undeclared identifier
      'io_iterator_t'
  io_service_t (*pIOIteratorNext)(io_iterator_t);
                                  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:197:3: warning: implicit declaration of function
      'CFTypeRef' is invalid in C99 [-Wimplicit-function-declaration]
  CFTypeRef (*pIORegistryEntryCreateCFProperty)(io_registry_entry_t,
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:197:15: error: use of undeclared identifier
      'pIORegistryEntryCreateCFProperty'
  CFTypeRef (*pIORegistryEntryCreateCFProperty)(io_registry_entry_t,
              ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:197:49: error: use of undeclared identifier
      'io_registry_entry_t'
  CFTypeRef (*pIORegistryEntryCreateCFProperty)(io_registry_entry_t,
                                                ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:198:49: error: use of undeclared identifier 'CFStringRef'
                                                CFStringRef,
                                                ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:199:49: error: use of undeclared identifier
      'CFAllocatorRef'
                                                CFAllocatorRef,
                                                ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:200:49: error: use of undeclared identifier 'IOOptionBits'
                                                IOOptionBits);
                                                ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:203:3: warning: implicit declaration of function
      'CFStringRef' is invalid in C99 [-Wimplicit-function-declaration]
  CFStringRef (*pCFStringCreateWithCString)(CFAllocatorRef,
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:203:17: error: use of undeclared identifier
      'pCFStringCreateWithCString'
  CFStringRef (*pCFStringCreateWithCString)(CFAllocatorRef,
                ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:204:45: error: expected expression
                                            const char*,
                                            ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:203:45: error: use of undeclared identifier
      'CFAllocatorRef'
  CFStringRef (*pCFStringCreateWithCString)(CFAllocatorRef,
                                            ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:205:45: error: use of undeclared identifier
      'CFStringEncoding'
                                            CFStringEncoding);
                                            ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:206:3: warning: implicit declaration of function
      'CFStringEncoding' is invalid in C99 [-Wimplicit-function-declaration]
  CFStringEncoding (*pCFStringGetSystemEncoding)(void);
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:206:22: error: use of undeclared identifier
      'pCFStringGetSystemEncoding'
  CFStringEncoding (*pCFStringGetSystemEncoding)(void);
                     ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:206:50: error: expected expression
  CFStringEncoding (*pCFStringGetSystemEncoding)(void);
                                                 ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:207:31: error: a parameter list without types is only
      allowed in a function definition
  UInt8 *(*pCFDataGetBytePtr)(CFDataRef);
                              ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:208:3: warning: implicit declaration of function 'CFIndex'
      is invalid in C99 [-Wimplicit-function-declaration]
  CFIndex (*pCFDataGetLength)(CFDataRef);
  ^
/Users/dylu/Documents/self-acc-c/third_party/libuv-1.41.0/src/unix/darwin.c:208:13: error: use of undeclared identifier
      'pCFDataGetLength'
  CFIndex (*pCFDataGetLength)(CFDataRef);
            ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
7 warnings and 20 errors generated.
```

**Comment by RaisinTen** [comment]
Hi. Could you please enclose the logs like this, so that it is easier to see?
````console
```console
paste the logs here
```
````
Also, what options are you using to compile libuv?

**Comment by vtjnash** [comment]
I am going to guess something is wrong with their build flags, since io_object_t is defined in darwin-stubs.h at the top of this file. But we can reopen if they reply.

**Comment by SpaceIm** [comment]
It has been fixed in 1.42.0: https://github.com/libuv/libuv/commit/1addf9b88a17bc32d009d377a14d540ccddd06db
